autoendpoint/src/routers/fcm/client.rs

```diff
@@ -238,10 +238,17 @@ impl FcmClient {
         // Handle error
         let status = response.status();
         if status.is_client_error() || status.is_server_error() {
-            let data: FcmResponse = response
-                .json()
+            let raw_data = response
+                .bytes()
                 .await
                 .map_err(FcmError::DeserializeResponse)?;
+            if raw_data.is_empty() {
+                return Err(FcmError::EmptyResponse.into());
+            }
+            let data: FcmResponse = serde_json::from_slice(&raw_data).map_err(|e| {
+                let s = String::from_utf8(raw_data.to_vec()).unwrap_or_else(|e| e.to_string());
```

Comment:
We probably shouldn't even bother including this string in the error until we're actually using it.

Reply:
Well, I was thinking that capturing it here would at least give us some clue. 
I mean, I'm fine logging it instead, but we don't really capture logs either.
I also suspect that this string probably isn't different between calls, so it might be less cardinal than we think?

(e.g. "Go away. This doesn't exist anymore" or whatever)

Reply:
Ok, also added `warn!()` messages. I figure Ops could enable them on a server and collect the logs. 

(hopefully by setting `RUST_LOG=autoendpoint=warn,error` for one of the boxes (if that's possible))

Reply:
Right, my point was these weren't going to output anywhere AFAICT, since they're not in the Error's `to_string`.

I created [a branch](https://github.com/mozilla-services/autopush-rs/compare/bug/sync-3931_fcm...bug/sync-3931_fcm-trait-extras?expand=1) with an alternative to the new log messages. Turns out it's not very difficult to get extras/tags out from the `Error` itself now that we have a trait for this purpose -- and this is part of the work we ultimately want for https://github.com/mozilla-services/autopush-rs/issues/367